### ISSUE TITLE
Brother HL1110

### ISSUE THREAD

**Issue by pdewacht** [pdewacht/brlaser#29]
Weird, GitHub deleted this issue. IIRC this printer had an odd 600x300 resolution mode, which is necessary to support, as this printer has very little RAM. Needs more investigation.

```sudo lpinfo --include-schemes usb -l -v
Device: uri = usb://Brother/HL-1110%20series?serial=dontlookhere
class = direct
info = Brother HL-1110 series
make-and-model = Brother HL-1110 series
device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-1110 series;CLS:PRINTER;CID:Brother Laser Type3;
location =```

**Comment by pali** [comment]
Hi! I have there printer labeled `brother model HL-11`. It should be `Brother HL-1112` and after running EXECUTE PRTCONFIG command (found in https://github.com/filisko/hl1110-utility-tool) I get printed test page with information:

```
HL-1110 series
CONT. ROM VERSION : 1.08
RAM SIZE : 1024Kbyte
```
(and lot of other printer settings)

I was trying to get this printer to work on Debian 9 Stretch system, but it was just printing empty pages. None CUPS driver for HL-10xx or HL-12xx worked.

But printer started working fine after selecting "DCP-7065DN, using brlaser v3 (en)" driver in CUPS menu.

Based on above series model, I guess that my HL-1112 is similar to HL-1110 in this issue. So both should work with "DCP-7065DN, using brlaser v3 (en)" driver.

Can you add HL-1110 series into supported list?

PS: EXECUTE PRTCONFIG command can be issued by following script (just replace IP_ADDR with print server address, or replace `nc` by `cat /dev/usb/lp0` for sending to local USB port).
```
{ echo -e "\e%-12345X@PJL\r\n@PJL EXECUTE PRTCONFIG \r\n\e%-12345X"; echo -e "\e"; } | nc IP_ADDR 9100
```

**Comment by pali** [comment]
When I connect it via USB, then lpinfo identify it as:
```
$ sudo lpinfo --include-schemes usb -l -v
Device: uri = usb://Brother/HL-1110%20series?serial=XXX
        class = direct
        info = Brother HL-1110 series
        make-and-model = Brother HL-1110 series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-1110 series;CLS:PRINTER;CID:Brother Laser Type3;
        location = 
```

**Comment by shaohua-deng** [comment]
@pali , thanks.

"DCP-7065DN, using brlaser v3 (en)", let my HL-1118 works on my ARM Debian 9 box.

Finally, I can set up a printer server at a very low power cost.

**Comment by pali** [comment]
@pdewacht Could you please include this printer to `brlaser.drv.in` file so it would be automatically detected by cups?

**Comment by DavidCussans** [comment]
HL-1110 prints test-page OK with brlaser v4 on Raspberry Pi

pi@raspberrypi:~ $ sudo lpinfo --include-schemes usb -l -v
Device: uri = usb://Brother/HL-1110%20series?serial=C4N818538
       class = direct
        info = Brother HL-1110 series
        make-and-model = Brother HL-1110 series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-1110 series;CLS:PRINTER;CID:Brother Laser Type3;
        location = 

ii  printer-driver-brlaser                                      4-1                                    armhf        printer driver for (some) Brother laser printers

However, I couldn't find HL-1110 in the list of models that CUPS listed when I installed the printer ( I picked "Brother DCP-1510 series, using brlaser (en)" , which seems to work)

 Many thanks for your efforts. Much appreciated.

**Comment by pali** [comment]
@QORTEC Could you add printer from this issue into definition list?

**Comment by QORTEC** [comment]
If I've read this issue correctly `HL-1110` functions correctly.

@pali I've added it to the list, thanks for pointing it out.

I'll probably push an update sometime July. It’s mainly dependent on how quickly I finish tasks I'm working on, and what my work schedule looks like.

I'm currently working tooling:
1. simplify retrieving the information for printer definitions, and output a correct definition for said printer. (finished)
2. validate printer definitions list to ensure all printers are correct and consistent
3. automate testing, then building (via solbuild?) before pushing to GitHub

Basically stuff to ensure quality, I've seen stuff occasionally fall through the cracks even with thorough review. Hopefully this will help fill in the gaps.

**Comment by QORTEC** [comment]
@pali I noticed that @pdewacht already defined the `HL-1110` in `brlaser.drv.in`. So the printer should show up in cups if you have the most recent version of `brlaser` installed.
I'll double check that the two `DeviceID`s are identical, though it should be.

**Edit**:
This printer was added to `brlaser` in `v5` of the driver.
`1284DeviceID` found in `brlaser.drv.in` is identical to the one you provided.
